lib/remote_console/server_adapter/websocket_uint8_utf8.rb

```diff
@@ -2,10 +2,11 @@ module RemoteConsole
   module ServerAdapter
     class WebsocketUint8Utf8 < WebsocketBinary
       def fetch(length)
```

Comment:
So, overall what is this method doing? It fetches data from a socket in a non-blocking fashion, and then returns a parsed version of the data which is a...hash?

Reply:
Nope, doesn't return but yields and it is a string. You cannot return with something that is asynchronous. Unfortunately you need this craziness due to the `WebSocket::Driver` :disappointed:

Reply:
Ok, so explain that in a comment please. :)